app/containers/popup.tsx

```diff
@@ -108,29 +90,32 @@ export class Popup extends React.Component<POPUP_PROPS & DispatchProps, {}> {
   };
 
   handleSwitch = () => {
-    if (this.props.isInterceptorOn[this.props.tabId]) {
+    if (this.props.data.isInterceptorOn) {
       this.props
-        .updateInterceptorStatus(this.props.tabId, false)
+        .updateInterceptorStatus(this.props.currentTab, false)
         .then(() => {
-          this.disableInterceptor(this.props.tabId);
-          this.updateBadgeIcon(this.props.tabId, true);
+          this.disableInterceptor(this.props.currentTab);
+          this.updateBadgeIcon(this.props.currentTab, true);
         })
-        .catch((err:any) => {
+        .catch((err: any) => {
           // something broke in the background store
           console.log(err);
         });
     } else {
-      this.props.updateInterceptorStatus(this.props.tabId, true);
-      this.updateBadgeIcon(this.props.tabId, false);
+      this.props.updateInterceptorStatus(this.props.currentTab, true);
+      this.updateBadgeIcon(this.props.currentTab, false);
     }
   };
 
   render() {
+    const props = this.props;
+    if (!props.data) {
+      return null;
```

Comment:
does this mean we end up showing a blank window under this condition?

Reply:
Yes, we dispatch an action with `tabId` as an argument. Whenever we open the `popup`, It takes some time(less than 2 sec) to reflect the `data.[tabId]` value in the store and ends up giving an error if this condition wasn't there. Until then, we show a blank window. And it only happens for the first time after the extension is freshly installed.

Reply:
okay. Try adding a loader or something (not a gif, but something like what FB does when loading the home page, for instance. May be in a later PR.